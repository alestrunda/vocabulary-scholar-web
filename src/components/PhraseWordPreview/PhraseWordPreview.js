import React from 'react';
import PropTypes from 'prop-types';
import FlatButton from 'material-ui/FlatButton';
import classNames from 'classnames';

const PhraseWordPreview = ({
  id,
  phrase,
  startIndex,
  endIndex,
  className,
  fillClassName,
  onEdit,
  onRemove,
  style,
  buttonStyle,
}) => {
  const handleRemoveBtnClick = () => {
    onRemove(id);
  };

  const handleEditBtnClick = () => {
    onEdit(id);
  };

  return (
    <div className={classNames('phrase-word', className)} style={style}>
      <p className="phrase-word__phrase">
        {phrase.substring(0, startIndex)}
        <span className={classNames('text-weight-bold', fillClassName)}>
          {phrase.substring(startIndex, endIndex)}
        </span>{' '}
        {phrase.substring(endIndex, phrase.length)}
      </p>
      <div className="phrase-word__action">
        {onEdit && (
          <FlatButton
            style={{
              minWidth: 70,
              height: 30,
              lineHeight: '30px',
              ...buttonStyle,
            }}
            labelStyle={{ fontSize: 12 }}
            className="mr10"
            primary
            data-testid="btn-edit"
            label="Edit"
            onClick={handleEditBtnClick}
          />
        )}
        {onRemove && (
          <FlatButton
            style={{
              minWidth: 70,
              height: 30,
              lineHeight: '30px',
              ...buttonStyle,
            }}
            labelStyle={{ fontSize: 12 }}
            data-testid="btn-remove"
            label="Remove"
            onClick={handleRemoveBtnClick}
          />
        )}
      </div>
    </div>
  );
};

PhraseWordPreview.propTypes = {
  id: PropTypes.string,
  className: PropTypes.string,
  fillClassName: PropTypes.string,
  startIndex: PropTypes.number,
  endIndex: PropTypes.number,
  phrase: PropTypes.string,
  onEdit: PropTypes.func,
  onRemove: PropTypes.func,
  style: PropTypes.object,
  buttonStyle: PropTypes.object,
};

export default PhraseWordPreview;
