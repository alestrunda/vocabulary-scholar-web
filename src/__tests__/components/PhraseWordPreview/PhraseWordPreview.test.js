import React from 'react';
import { shallow } from 'enzyme';

import PhraseWordPreview from '../../../components/PhraseWordPreview';

it('renders correctly', () => {
  const phrase = {
    id: 'abc123',
    phrase: 'Lorem ipsum dolor',
    startIndex: 6,
    endIndex: 11,
  };
  const root = shallow(<PhraseWordPreview {...phrase} />);
  expect(root).toMatchSnapshot();
});

it('triggers onEdit', () => {
  const phrase = {
    id: 'abc123',
    phrase: 'Lorem ipsum dolor',
    startIndex: 6,
    endIndex: 11,
  };
  const onEdit = jest.fn();
  const root = shallow(<PhraseWordPreview onEdit={onEdit} {...phrase} />);
  root.find('[data-testid="btn-edit"]').simulate('click');
  expect(onEdit).toHaveBeenCalledWith(phrase.id);
});

it('triggers onRemove', () => {
  const phrase = {
    id: 'abc123',
    phrase: 'Lorem ipsum dolor',
    startIndex: 6,
    endIndex: 11,
  };
  const onRemove = jest.fn();
  const root = shallow(<PhraseWordPreview onRemove={onRemove} {...phrase} />);
  root.find('[data-testid="btn-remove"]').simulate('click');
  expect(onRemove).toHaveBeenCalledWith(phrase.id);
});
