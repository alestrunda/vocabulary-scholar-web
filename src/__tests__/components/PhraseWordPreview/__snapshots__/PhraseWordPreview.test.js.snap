// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PhraseWordPreview
    endIndex={11}
    id="abc123"
    phrase="Lorem ipsum dolor"
    startIndex={6}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="phrase-word__phrase"
        >
          Lorem 
          <span
            className="text-weight-bold"
          >
            ipsum
          </span>
           
           dolor
        </p>,
        <div
          className="phrase-word__action"
        />,
      ],
      "className": "phrase-word",
      "style": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Lorem ",
            <span
              className="text-weight-bold"
            >
              ipsum
            </span>,
            " ",
            " dolor",
          ],
          "className": "phrase-word__phrase",
        },
        "ref": null,
        "rendered": Array [
          "Lorem ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "ipsum",
              "className": "text-weight-bold",
            },
            "ref": null,
            "rendered": "ipsum",
            "type": "span",
          },
          " ",
          " dolor",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            undefined,
          ],
          "className": "phrase-word__action",
        },
        "ref": null,
        "rendered": Array [
          undefined,
          undefined,
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="phrase-word__phrase"
          >
            Lorem 
            <span
              className="text-weight-bold"
            >
              ipsum
            </span>
             
             dolor
          </p>,
          <div
            className="phrase-word__action"
          />,
        ],
        "className": "phrase-word",
        "style": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Lorem ",
              <span
                className="text-weight-bold"
              >
                ipsum
              </span>,
              " ",
              " dolor",
            ],
            "className": "phrase-word__phrase",
          },
          "ref": null,
          "rendered": Array [
            "Lorem ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "ipsum",
                "className": "text-weight-bold",
              },
              "ref": null,
              "rendered": "ipsum",
              "type": "span",
            },
            " ",
            " dolor",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              undefined,
            ],
            "className": "phrase-word__action",
          },
          "ref": null,
          "rendered": Array [
            undefined,
            undefined,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
